Makefile.am

```diff
@@ -38,6 +38,8 @@ AM_CPPFLAGS = \
 	-include $(top_builddir)/config.h \
 	-I$(top_builddir) \
 	-I$(top_srcdir) \
+	-I$(top_srcdir)/src/mavlink-router \
+	-I$(top_srcdir)/src/common \
```

Comment:
Code move on a separated PR, with this one on top: easy.
Properly extract mainloop: not so...(way easy just explain on commit message =P )

Can I do just the first now? =D